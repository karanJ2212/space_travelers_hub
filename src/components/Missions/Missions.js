import { useSelector } from "react-redux";

export default function Missions() {
  const missions = useSelector((store) => store.missions.missions || []);
  return (
    <div>
      <table>
        <thead>
          <tr>
            <th>Mission</th>
            <th>Description</th>
            <th>Status</th>
            <th></th>
          </tr>
        </thead>
        <tbody>
          {missions.map((mission) => (
            <tr key={mission.mission_id}>
              <td>{mission.mission_name}</td>
              <td>{mission.mission_description}</td>
              <td></td>
              <td></td>
            </tr>
          ))}
        </tbody>
      </table>
    </div>
  );
}
