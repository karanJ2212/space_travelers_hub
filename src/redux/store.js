import { configureStore } from '@reduxjs/toolkit';
import rocketReducer from './rockets/rocketsSlice';
import missionReducer from './missions/missionslice';

const store = configureStore({
  reducer: {
    rockets: rocketReducer,
    missions: missionReducer,
  },
});

export default store;
