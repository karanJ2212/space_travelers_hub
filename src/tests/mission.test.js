import { Provider } from "react-redux";
import { render } from "@testing-library/react";
import Missions from "../components/Missions/Missions";
import store from "../redux/store";
import {
  getMissions,
  joinMission,
  leaveMission,
} from "../redux/missions/missionslice";

describe("display the mission in the page", () => {
  it("should display all the missions", () => {
    const { container } = render(
      <Provider store={store}>
        <Missions />
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });
});
describe("Missions reducers", () => {
  it("should fetch missions", async () => {
    await store.dispatch(getMissions());
    expect(store.getState().missions.missions.length).toEqual(10);
  });
  it("should join a mission", () => {
    store.dispatch(joinMission("9D1B7E0"));
    expect(store.getState().missions.missions[0].reserved).toBe(true);
  });
  it("should leave a mission that has been joined", () => {
    store.dispatch(leaveMission("9D1B7E0"));
    expect(store.getState().missions.missions[0].reserved).toEqual(false);
  });
});
