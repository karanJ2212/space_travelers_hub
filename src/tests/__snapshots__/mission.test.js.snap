// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`display the mission in the page should display all the missions 1`] = `
<div>
  <div
    class="loading"
  >
    Loading...
  </div>
</div>
`;
